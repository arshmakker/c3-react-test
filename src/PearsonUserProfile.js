import React from "react";
import styles from "./PearsonUserProfile.module.css";

export class PearsonUserProfile extends React.Component {
  _deleteUser = () => {
    this.props.deleteUser(this.props.id);
  };
  render() {
    return (
      <div className={styles.pearsonUserProfileBox}>
        <img
          className={styles.pearsonUserProfileAvatar}
          src={this.props.avatar}
          alt="User Profile"
        />
        <div className={styles.pearsonUserName}>
          {this.props.first_name} {this.props.last_name}
        </div>
        <div onClick={this._deleteUser} className={styles.personUserButtons}>
          Delete
        </div>
      </div>
    );
  }
}
