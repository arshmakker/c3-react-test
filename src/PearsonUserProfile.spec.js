import React from "react";
import { shallow } from "enzyme";
import { PearsonUserProfile } from "./PearsonUserProfile";
import renderer from "react-test-renderer";

describe("PearsonUserProfile", () => {
  let component;

  beforeEach(() => {
    component = shallow(<PearsonUserProfile />);
  });

  it("renders three divs", () => {
    expect(component.find("div")).toHaveLength(3);
  });

  it("renders an img tag for Avatar", () => {
    expect(component.find("img")).toHaveLength(1);
  });

  it("renders Pearson User Profile as per the snapshot", () => {
    const tree = renderer
      .create(
        <PearsonUserProfile
          key="1"
          id="1"
          first_name="Eve"
          last_name="Holt"
          avatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
