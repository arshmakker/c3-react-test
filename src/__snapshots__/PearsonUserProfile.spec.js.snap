// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PearsonUserProfile renders Pearson User Profile as per the snapshot 1`] = `
<div
  className="pearsonUserProfileBox"
>
  <img
    alt="User Profile"
    className="pearsonUserProfileAvatar"
    src="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
  />
  <div
    className="pearsonUserName"
  >
    Eve
     
    Holt
  </div>
  <div
    className="personUserButtons"
    onClick={[Function]}
  >
    Delete
  </div>
</div>
`;
